parse.go

```diff
@@ -527,7 +531,9 @@ func (p *Parser) process(args []string) error {
 
 			// instantiate the field to point to a new struct
 			v := p.val(subcmd.dest)
-			v.Set(reflect.New(v.Type().Elem())) // we already checked that all subcommands are struct pointers
```

Comment:
This line is so that when I write `./myprogram subcmd` then the field corresponding to `subcmd` is instantiated to a non-nil value. Why should this behavior change when ignoring default values?

Reply:
The behavior is identical as before when `IgnoreDefault` is false, as well as when the structure is still `nil`.
The goal of not overriding an existing non-nil value is to allow an existing struct to retain its original values.

My specific use-case is:
1. Calling `arg.NewParser(...).Parse(...)` once with CLI arguments, which can include a `--config-file=some.json` argument. At this point the structure is populated with default values, environment values, or CLI arguments - in ascending order of priority.
2. Then the parsed structure is enriched by unmarshaling the config file.
3. The json took priority over the default, and environment values, but also the CLI flags. So I re-parse the CLI flags, but ignore environment and default values.

All this to have values be set by CLI > JSON > default > environment - in this order of priority.

Reply:
Sorry for the delay in getting to this. I understand your logic now. Can you make the condition just `if v.IsNil()`? I'd rather keep the logic as simple as possible for this.

Reply:
Yeah sure. I intended not to change the existing behavior when the flag is not set, but simpler is indeed better 👍🏻